bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -210,23 +211,17 @@ public static String getBroadcastAddress() {
 
         while (en.hasMoreElements()) {
             NetworkInterface networkInterface = en.nextElement();
-            boolean isLoopback = true;
+
             try {
-                isLoopback = networkInterface.isLoopback();
+                if (!networkInterface.isUp() || networkInterface.isLoopback()) {
+                    continue;
+                }
             } catch (SocketException ignored) {
-            }
-            if (isLoopback) {
                 continue;
             }
-            for (Iterator<InterfaceAddress> it = networkInterface.getInterfaceAddresses().iterator(); it.hasNext();) {
-                InterfaceAddress interfaceAddress = it.next();
-                if (addPrefix) {
-                    interfaceIPs.add(interfaceAddress.getAddress().getHostAddress() + "/"
-                            + interfaceAddress.getNetworkPrefixLength());
-                } else {
-                    interfaceIPs.add(interfaceAddress.getAddress().getHostAddress());
-                }
 
+            for (InterfaceAddress cidr : networkInterface.getInterfaceAddresses()) {
+                interfaceIPs.add(new CidrAddress(cidr.getAddress(), cidr.getNetworkPrefixLength()));
```

Comment:
You could remote `@SuppressWarnings("null")` for the whole method implementation using e.g.

```Java
final InetAddress address = cidr.getAddress();
assert address != null; // NetworkInterface.getInterfaceAddresses() should return only non-null addresses
interfaceIPs.add(new CidrAddress(address, cidr.getNetworkPrefixLength()));
```

WDYT? I am not a friend of supress warnings for the whole function. An assert should not change anything on runtime as they are default deactivated.

Reply:
I agree